py/server/deephaven/_table_reader.py

```diff
@@ -83,3 +108,98 @@ def _table_reader_chunks(table: Table, *, cols: Optional[Union[str, Sequence[str
             j_reader_context.close()
             row_sequence_iterator.close()
 
+
+def _table_reader_dict(table: Table, cols: Optional[Union[str, Sequence[str]]] = None, chunk_size: int = 2048) \
+        -> Generator[Dict[str, Any], None, None]:
+    """ A generator that reads one row at a time from a table into a dictionary. The dictionary is a map of column names
+    to scalar values of the column data type.
+
+    Args:
+        table (Table): The table to read.
+        cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read.
+        chunk_size (int): The number of rows to read at a time internally to reduce the number of Java/Python boundary
+            crossings. Default is 2048.
+
+    Returns:
+        A generator that yields a dictionary of column names to a value.
```

Comment:
```suggestion
        A generator that yields a dictionary of column names to a values.
```
See if this needs to go anywhere else.